### ISSUE TITLE
Janssen's Notes for Tutorial 2 Review

### ISSUE THREAD

**Issue by lyf-woods** [prajogt/ttc_bus_delays#1]
# Notes by Janssen Myer Rambaud, reviewing Timothius Prajogi for STA302 Mini-Essay 2

I would suggest you add the preamble comments at the top for extra documentation as done in the Telling Stories with Data - 2 chapter. 
The template given to us in the book is as follows:
> Preamble 
> Purpose: Read in data from the 2022 Australian Election and make
> a graph of the number of seats each party won.
> Author: Rohan Alexander
> Email: rohan.alexander@utoronto.ca
> Date: 1 January 2023
> Prerequisites: Know where to get Australian elections data.

I would add this preamble to your:
- 00-retrieve_data.R
- 01-data_cleaning.R
- ttc_bus_delay.qmd
files just for extra documentation and added context for the reader.

I like the organization scheme of your repo, everything is tidy if we follow the format given to us in the textbook

## 00-retrieve_data.R

```
ttc_bus_delay_package <- search_packages("TTC Bus Delay Data")
ttc_bus_delay_resource <- ttc_bus_delay_package |> list_package_resources()
ttc_bus_delay_statistics <- 
  ttc_bus_delay_resource |> 
  dplyr::filter(row_number()==n()) |> 
  get_resource()
```

I believe this part is somewhat redundant and you should be able to just pipe it continuously without having to declare new variables. It will make things easier to trace in your code because of the increased simplicity and it will also lessen the amount of 'data' we have in our workspace.

## 01-data_cleaning.R

```# Remove irrelevant rows
cleaned_ttc_delay_data <- cleaned_ttc_delay_data |> 
  select(
    date, 
    route,
    time, 
    day, 
    # location removed since values varied too much to derive relationships
    # (~11,000 unique values for ~50,000 entries)
    # additionally, route provides similar info
    incident, 
    min_delay
    # min_gap removed since min_delay is the value we are interested in
    # direction removed since it is irrelevant
    # vehicle removed since it is irrelevant
  )
```

This was a nice touch because it shows the original format of the downloaded dataset and what you took out to clean, and why. This would be a change that I should incorporate in my future work because I just selected the important elements without really diving into detail what the original dataset had.

## ttc_bus_delay.qmd
The organization and style decisions here were very well done. Each section is separated into a proper flowing process, and you even included label/figure captions for the graphs. This is something I should have done too for my part.

There are some minor grammar mistakes, like:
> This includes of categories...
> The **staistics** for the TTC...
But this is just picky observation and does not reflect the quality of the R code and report.

The code works and when running it, I get these two graphs:
![image](https://github.com/prajogt/ttc_bus_delays/assets/58833849/eab1b7bb-bdec-4d1c-952f-dd987a55e96e)
![image](https://github.com/prajogt/ttc_bus_delays/assets/58833849/89579e88-079a-4b2b-a7b6-712a028bd36e)

If you are considering their technical presentation, then they do well and portray all the information needed. One can easily look and analyze the graphs themselves and come up with their own observations, but if you consider these graphs from a design perspective, then they are quite bland. They are strictly black and white and seem very dull. This is not to say that research and statistical work should be very bright and consist of every colour in the world, but a simple touch of colour may elevate your graphs to higher levels. You can use different colours for each bar, and maybe a single colour like blue, green, red, or orange on the "Amount of delays as a function of time" graph in order to make it pop a bit more.

## Conclusion
Overall this work was very well done. You retrieved an interesting dataset, cleaned it properly, and presented the graphs to provide the reader with an informative experience. There were some things I found that could have been improved, but they are more so grammatical and documentation concerns.
